app/controllers/api/v1/packages_controller.rb

```diff
@@ -88,17 +94,54 @@ def add_item_to_stockit
     def package_params
       attributes = [:quantity, :length, :width, :height, :notes, :item_id,
         :received_at, :rejected_at, :package_type_id, :state_event, :image_id,
-        :inventory_number, :location_id]
+        :inventory_number, :location_id, :designation_name]
       params.require(:package).permit(attributes)
     end
 
+    def item_attributes
+      if (item = item_params)
+        item["donor_condition_id"] = DonorCondition.find_by(name_en: item["donor_condition_id"]).try(:id)
+        item
+      end
+    end
+
+    def item_params
+      params["package"].require(:item).permit(:donor_condition_id)
+    end
+
+    def save_item_details
+      params["package"]["item"] &&
+      (attributes = item_attributes) &&
+      @package.item.update_attributes(attributes)
+    end
+
     def serializer
       Api::V1::PackageSerializer
     end
 
     def offer_id
-      Item.where(id: @package.item_id).pluck(:offer_id).first
+      @package.item.offer_id
     end
 
+    def package_record
+      if package_params[:inventory_number]
+        if existing_package
+          GoodcitySync.request_from_stockit = true
+          @package.assign_attributes(package_params)
+          @package.location_id = location_id
+          @package
+        end
+      else
+        @package = Package.new(package_params)
+      end
+    end
+
+    def location_id
+      Location.find_by(stockit_id: package_params[:location_id]).try(:id)
+    end
+
+    def existing_package
+      @package = Package.find_by(inventory_number: package_params[:inventory_number])
+    end
```

Comment:
I found I don't really understand what this method does from the name `existing_package`. I don't mind too much if it's separate from the `existing_package` method but it's hard to understand the logic flow of `existing_package` without having to actually read what this method does. A better name would be `find_package_by_inventory_number` and then it would make sense in the logic flow of `existing_package`

However, since this is a one-line method, I would be tempted to remove it anyway and save the call stack. I find one-line methods can make the code harder to read and understand. They can be justified if they are called in several places but my brain only has a limited stack!!